executable/microsoft_pe.ksy

```diff
@@ -355,6 +355,11 @@ types:
       body:
         pos: pointer_to_raw_data
         size: size_of_raw_data
+      resource_table:
```

Comment:
Maybe it would have been better is to put these fields under `data_dirs.resource_table`, although I wasn't sure how to create a dependency on parsing each sections *first* then adding the fields, e.g. something like:
```
  data_dir:                                                                     
    seq:                                                                        
      - id: virtual_address                                                     
        type: u4                                                                
      - id: size                                                                
        type: u4                                                                
  instances:
    resource_table:
      type: resource_directory_table(0, _root.pe.sections[x].pointer_to_raw_data, _root.pe.sections[x].virtual_address)
      if: virtual_address == _root.pe.sections[x].virtual_address
```
where `x` would be an implicit iterator.

Reply:
Instances are parsed lazily. So dependencies are tracked: on each access go an insfance it is checked if it js initialized, and if not, it is tried to be initialized.

Reply:
The issue I'm facing with this solution would be to iterate over the `_root.pe.sections`, find which one has the same virtual-address in order to create the instance.

In imperative pseudo-code:
```
instance = null;
for section in _root.pe.sections:
  if virtual_address == section.virtual_address:
    instance = resource_directory_table(0, section.pointer_to_raw_data, section.virtual_address))
return instance
```

I was thinking of something like this:
```
      type: resource_directory_table(0, _root.pe.sections[x].pointer_to_raw_data, _root.pe.sections[x].virtual_address)
      if: virtual_address == _root.pe.sections[x].virtual_address
```
where `x` would be an implicit iterator.

Otherwise the instance has to be put into `sections` and will be created once parsing each sections will result in the condition only if the virtual address being true.

Reply:
Stateful parsing is not supported out of the box, but if you can reformulate the problem as 2 stages, 1 is iterating using `repeat: until`, and the second is taking its last element, it is possible to do something. You may want to look at https://github.com/kaitai-io/kaitai_struct_formats/pull/265/files#diff-450d8347f68fb61616750e47db6a9f4bR315